covariancematrix.go

```diff
@@ -80,3 +80,71 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 	cov.Scale(1/(n-1), cov)
 	return cov
 }
+
+// CorrelationMatrix calculates a correlation matrix from a matrix of data,
+// using a two-pass algorithm. The matrix returned will be symmetric, square,
+// and positive-semidefinite.
+//
+// The weights wts should have the length equal to the number of rows in
+// input data matrix x. cov should either be a square matrix with the same
```

Comment:
How about I just call it `c`?  The variable starts out as covariance, and then ends as correlation.  I made it `cov` so that the input parameters would be more descriptive.